### DISCUSSION TITLE
How to change plot width for a Julia plot?

### DISCUSSION THREAD

**Discussion by sje30** [quarto-dev/quarto-cli#3138] [Q&A]
Hi,

I have the following example that I compile with `quarto render f.qmd`

````
---
title: "test plot"
author: "SJE"
format:
  beamer:
    keep-tex: true
---

## Test slide

* this is a test slide

* with some points 

* example plot below

  
```{julia} 
#| fig-width: 1
using Plots
plot([1:10])
```

````

I'd like to make the plot smaller so that it fits on the page, but what I get is a plot that overflows:
<img width="652" alt="Screenshot 2022-11-01 at 22 41 24" src="https://user-images.githubusercontent.com/1095067/199355669-57c02715-2df0-4ba8-a92a-9198aaaeb53b.png">

My workaround for now is to do something like this in the yaml header:

```
format:
  beamer:
    header-includes: |
      \setkeys{Gin}{width=8cm,height=6cm,keepaspectratio}
```

but that obviously sets the figure size for all figures.

**Comment by sje30** [comment]
p.s. examples of these can be found at:
https://github.com/Nick-Gale/2022_Scientific_Programming_CamCompBio/tree/main/Lectures

see lecture 4 for examples of plotting.

**Reply by sje30** [reply]
possibly related: https://github.com/rstudio/rstudio/issues/11708. (for R code)

**Reply by cscheid** [reply]
Does

````
```{julia}
#| fig-width: 4in
...
```
````

not work?

**Comment by sje30** [comment]
not for me, I'm afraid.  Here's what I tried, and the resulting .tex snippet and pdf -- no sign of images being rescaled.

````
---
title: "test plot"
author: "SJE"
format:
  beamer:
    keep-tex: true
---

## Test slide

* this is a test slide

* with some points 

* example plot, trying fig-width: 1in

  
```{julia} 
#| fig-width: 1in
using Plots
plot([1:10])
```

````

````
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}

...


\begin{frame}{Test slide}
\protect\hypertarget{test-slide}{}
\begin{itemize}
\item
  this is a test slide
\item
  with some points
\item
  example plot, trying fig-width: 1in
\end{itemize}

\includegraphics{t2_files/figure-beamer/cell-2-output-1.pdf}
\end{frame}

````
<img width="634" alt="Screenshot 2022-11-02 at 19 49 32" src="https://user-images.githubusercontent.com/1095067/199588597-f270ce8c-44bc-4089-bd6b-179d7b7984b0.png">

**Reply by cscheid** [reply]
Ok. I can reproduce this locally. I don't understand what's going wrong, though, because we _are_ supposed to be communicating widths and heights to julia in `setup.jl`.

In the meantime, you can force the sizing computation yourself by passing a size parameter to `plot()`; I believe it takes pixels as the size, and PDF assumes 96 ppi. So

```julia
plot([1:10], (192, 192))
```

would make a 2in x 2in plot.

**Reply by sje30** [reply]
Thanks, this is a good workaround, and works for me.

````
---
title: "test plot"
author: "SJE"
format:
  beamer:
    keep-tex: true
---

## Test slide

* this is a test slide

* with some points 

* example plot, with explicit size=(a,b)

  
```{julia} 
using Plots
plot([1:10], size=(200,100))
```

````
<img width="548" alt="Screenshot 2022-11-03 at 08 36 38" src="https://user-images.githubusercontent.com/1095067/199677471-00d24be7-0430-4d2b-828d-78df4a203f91.png">

**Reply by cscheid** [reply]
Ok, so this workaround turns out to be the currently recommended solution for the entire jupyter ecosystem. You can still configure width and height on a _global_ level in the YAML metadata. But per-cell configuration is not supported in `julia` or `python` cells. See #3155.
